### DISCUSSION TITLE
`yamlValidation` contribution point

### DISCUSSION THREAD

**Discussion by kizivat** [redhat-developer/vscode-yaml#776] [Q&A]
My primary goal is that my own VS code extension validates certain YAML files using a schema.

Firstly I tried to set up a language client using [`yaml-language-server`](https://github.com/redhat-developer/yaml-language-server) and using `defaultConfiguration` contribution point to set default value `yaml.schemas`. What I realized was that the user of my extension would need to have the file with the schema definition somehow distributed to them so I dropped this approach for now.

Then, I found out about `yamlValidation` contribution point. I'm not sure how this should be used though. Is it possible to use without asking users to install your extension (meaning `redhat.vscode-yaml`)?

However, for now I tried to set up the contribution point,

```json
"contributes": {
  "yamlValidation": [
    {
      "fileMatch":"**/myPattern/**/*.{yml,yaml}",
      "url": "./my-schema.json"
    }
  ]
```

put `my-schema.json` into the root of the extension, launched the extension… But it's not working. In this case, I have had `redhat.vscode-yaml` installed in the instance of VS Code I tested it in.

I'm not sure what I'm missing.

**Comment by gorkem** [comment]
Are you including `redhat.vscode-yaml` on the `extensionDependencies` section of the [extension manifest](https://code.visualstudio.com/api/references/extension-manifest) this should cause the vscode-yaml to be installed automatically.
